Compilation warnings pending cleanup

-- Found PythonInterp: /root/miniconda3/bin/python3 (found suitable version "3.10.18", minimum required
-- Found PythonLibs: /root/miniconda3/lib/libpython3.10.so
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- Found pybind11: /root/miniconda3/lib/python3.10/site-packages/pybind11/include (found version "2.13.6
-- TORCH_DIR = /root/miniconda3/lib/python3.10/site-packages/torch
-- TORCH_NPU_DIR = /root/miniconda3/lib/python3.10/site-packages/torch_npu
-- Attempting to determine PyTorch C++ ABI compatibility...
-- PyTorch reports old C++ ABI (False). Setting _GLIBCXX_USE_CXX11_ABI=0.
-- Final determined _GLIBCXX_USE_CXX11_ABI=0
-- CMAKE_PREFIX_PATH = /root/miniconda3/lib/python3.10/site-packages/pybind11;/root/miniconda3/lib/pytho
-- ================
-- TARGET_INSTALL_DIR = /home/x00575210/sgl-kernel-npu/output
-- Configuring done (7.1s)
-- Generating done (0.0s)
-- Build files have been written to: /home/x00575210/sgl-kernel-npu/csrc/build
/usr/local/cmake-3.31.3/bin/cmake -S/home/x00575210/sgl-kernel-npu/csrc -B/home/x00575210/sgl-kernel-npu
/usr/local/cmake-3.31.3/bin/cmake -E cmake_progress_start /home/x00575210/sgl-kernel-npu/csrc/build/CMak
/usr/bin/gmake  -f CMakeFiles/Makefile2 all
gmake[1]: Entering directory '/home/x00575210/sgl-kernel-npu/csrc/build'
/usr/bin/gmake  -f CMakeFiles/deep_ep_cpp.dir/build.make CMakeFiles/deep_ep_cpp.dir/depend
gmake[2]: Entering directory '/home/x00575210/sgl-kernel-npu/csrc/build'
cd /home/x00575210/sgl-kernel-npu/csrc/build && /usr/local/cmake-3.31.3/bin/cmake -E cmake_depends "Unixl-npu/csrc/build /home/x00575210/sgl-kernel-npu/csrc/build /home/x00575210/sgl-kernel-npu/csrc/build/CMa
gmake[2]: Leaving directory '/home/x00575210/sgl-kernel-npu/csrc/build'
/usr/bin/gmake  -f CMakeFiles/deep_ep_cpp.dir/build.make CMakeFiles/deep_ep_cpp.dir/build
gmake[2]: Entering directory '/home/x00575210/sgl-kernel-npu/csrc/build'
[ 25%] Building CXX object CMakeFiles/deep_ep_cpp.dir/pybind_extension.cpp.o
/usr/bin/c++ -Ddeep_ep_cpp_EXPORTS -I/home/x00575210/sgl-kernel-npu/csrc/csrc -I/home/x00575210/sgl-kernpython3.10/site-packages/torch/include/torch/csrc/api/include -I/root/miniconda3/lib/python3.10/site-pac/acl -I/root/miniconda3/lib/python3.10/site-packages/torch_npu/include/third_party/acl/inc -I/usr/local/-isystem /root/miniconda3/include/python3.10 -O3 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -std=gnu++17 -fPIC -fvkeFiles/deep_ep_cpp.dir/pybind_extension.cpp.o.d -o CMakeFiles/deep_ep_cpp.dir/pybind_extension.cpp.o -c
[ 50%] Building CXX object CMakeFiles/deep_ep_cpp.dir/deepep/config.cpp.o
[ 75%] Building CXX object CMakeFiles/deep_ep_cpp.dir/deepep/deep_ep.cpp.o
/usr/bin/c++ -Ddeep_ep_cpp_EXPORTS -I/home/x00575210/sgl-kernel-npu/csrc/csrc -I/home/x00575210/sgl-kernpython3.10/site-packages/torch/include/torch/csrc/api/include -I/root/miniconda3/lib/python3.10/site-pac/acl -I/root/miniconda3/lib/python3.10/site-packages/torch_npu/include/third_party/acl/inc -I/usr/local/-isystem /root/miniconda3/include/python3.10 -O3 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -std=gnu++17 -fPIC -fviles/deep_ep_cpp.dir/deepep/config.cpp.o.d -o CMakeFiles/deep_ep_cpp.dir/deepep/config.cpp.o -c /home/x0
/usr/bin/c++ -Ddeep_ep_cpp_EXPORTS -I/home/x00575210/sgl-kernel-npu/csrc/csrc -I/home/x00575210/sgl-kernpython3.10/site-packages/torch/include/torch/csrc/api/include -I/root/miniconda3/lib/python3.10/site-pac/acl -I/root/miniconda3/lib/python3.10/site-packages/torch_npu/include/third_party/acl/inc -I/usr/local/-isystem /root/miniconda3/include/python3.10 -O3 -fPIC -D_GLIBCXX_USE_CXX11_ABI=0 -std=gnu++17 -fPIC -fvFiles/deep_ep_cpp.dir/deepep/deep_ep.cpp.o.d -o CMakeFiles/deep_ep_cpp.dir/deepep/deep_ep.cpp.o -c /home
In file included from /home/x00575210/sgl-kernel-npu/csrc/deepep/deep_ep.cpp:7:
/home/x00575210/sgl-kernel-npu/csrc/deepep/deep_ep.cpp: In member function ‘std::tuple<at::Tensor, std::std::function<void()> > > deep_ep::Buffer::low_latency_dispatch(const at::Tensor&, const at::Tensor&, co
/home/x00575210/sgl-kernel-npu/csrc/deepep/pytorch_npu_helper.hpp:543:22: warning: narrowing conversion
  543 |           at::empty({workspace_size}, options.dtype(c10::kByte));             \
      |                      ^~~~~~~~~~~~~~
/home/x00575210/sgl-kernel-npu/csrc/deepep/pytorch_npu_helper.hpp:543:22: note: in definition of macro ‘
  543 |           at::empty({workspace_size}, options.dtype(c10::kByte));             \
      |                      ^~~~~~~~~~~~~~
/home/x00575210/sgl-kernel-npu/csrc/deepep/pytorch_npu_helper.hpp:543:22: warning: narrowing conversion
  543 |           at::empty({workspace_size}, options.dtype(c10::kByte));             \
      |                      ^~~~~~~~~~~~~~
/home/x00575210/sgl-kernel-npu/csrc/deepep/pytorch_npu_helper.hpp:543:22: note: in definition of macro ‘
  543 |           at::empty({workspace_size}, options.dtype(c10::kByte));             \
      |                      ^~~~~~~~~~~~~~
/home/x00575210/sgl-kernel-npu/csrc/deepep/deep_ep.cpp: In member function ‘std::tuple<at::Tensor, std::const at::Tensor&, const at::Tensor&, const at::Tensor&, const at::Tensor&, const at::Tensor&, int64_t,
/home/x00575210/sgl-kernel-npu/csrc/deepep/pytorch_npu_helper.hpp:543:22: warning: narrowing conversion
  543 |           at::empty({workspace_size}, options.dtype(c10::kByte));             \
      |                      ^~~~~~~~~~~~~~
/home/x00575210/sgl-kernel-npu/csrc/deepep/pytorch_npu_helper.hpp:543:22: note: in definition of macro ‘
  543 |           at::empty({workspace_size}, options.dtype(c10::kByte));             \
      |                      ^~~~~~~~~~~~~~
/home/x00575210/sgl-kernel-npu/csrc/deepep/pytorch_npu_helper.hpp:543:22: warning: narrowing conversion
  543 |           at::empty({workspace_size}, options.dtype(c10::kByte));             \
      |                      ^~~~~~~~~~~~~~
/home/x00575210/sgl-kernel-npu/csrc/deepep/pytorch_npu_helper.hpp:543:22: note: in definition of macro ‘
  543 |           at::empty({workspace_size}, options.dtype(c10::kByte));

--- a/csrc/deepep/pytorch_npu_helper.hpp
+++ b/csrc/deepep/pytorch_npu_helper.hpp
@@ -517,55 +517,55 @@ typedef int (*InitHugeMemThreadLocal)(void *, bool);
 typedef void (*UnInitHugeMemThreadLocal)(void *, bool);
 typedef void (*ReleaseHugeMem)(void *, bool);
 
-#define EXEC_NPU_CMD(aclnn_api, ...)                                                                          \
-    do {                                                                                                      \
-        static const auto getWorkspaceSizeFuncAddr = GetOpApiFuncAddr(#aclnn_api "GetWorkspaceSize");         \
-        static const auto opApiFuncAddr = GetOpApiFuncAddr(#aclnn_api);                                       \
-        static const auto initMemAddr = GetOpApiFuncAddr("InitHugeMemThreadLocal");                           \
-        static const auto unInitMemAddr = GetOpApiFuncAddr("UnInitHugeMemThreadLocal");                       \
-        static const auto releaseMemAddr = GetOpApiFuncAddr("ReleaseHugeMem");                                \
-        TORCH_CHECK(getWorkspaceSizeFuncAddr != nullptr && opApiFuncAddr != nullptr, #aclnn_api, " or ",      \
-                    #aclnn_api "GetWorkspaceSize", " not in ", GetOpApiLibName(), ", or ", GetOpApiLibName(), \
-                    "not found.");                                                                            \
-        auto acl_stream = c10_npu::getCurrentNPUStream().stream(false);                                       \
-        uint64_t workspace_size = 0;                                                                          \
-        uint64_t *workspace_size_addr = &workspace_size;                                                      \
-        aclOpExecutor *executor = nullptr;                                                                    \
-        aclOpExecutor **executor_addr = &executor;                                                            \
-        InitHugeMemThreadLocal initMemFunc = reinterpret_cast<InitHugeMemThreadLocal>(initMemAddr);           \
-        UnInitHugeMemThreadLocal unInitMemFunc = reinterpret_cast<UnInitHugeMemThreadLocal>(unInitMemAddr);   \
-        if (initMemFunc) {                                                                                    \
-            initMemFunc(nullptr, false);                                                                      \
-        }                                                                                                     \
-        auto converted_params = ConvertTypes(__VA_ARGS__, workspace_size_addr, executor_addr);                \
-        static auto getWorkspaceSizeFunc = ConvertToOpApiFunc(converted_params, getWorkspaceSizeFuncAddr);    \
-        auto workspace_status = call(getWorkspaceSizeFunc, converted_params);                                 \
-        TORCH_CHECK(workspace_status == 0, "call " #aclnn_api " failed, detail:", aclGetRecentErrMsg());      \
-        void *workspace_addr = nullptr;                                                                       \
-        if (workspace_size != 0) {                                                                            \
-            at::TensorOptions options = at::TensorOptions(torch_npu::utils::get_npu_device_type());           \
-            auto workspace_tensor = at::empty({workspace_size}, options.dtype(c10::kByte));                   \
-            workspace_addr = const_cast<void *>(workspace_tensor.storage().data());                           \
-        }                                                                                                     \
-        auto acl_call = [converted_params, workspace_addr, workspace_size, acl_stream, executor]() -> int {   \
-            typedef int (*OpApiFunc)(void *, uint64_t, aclOpExecutor *, const aclrtStream);                   \
-            OpApiFunc opApiFunc = reinterpret_cast<OpApiFunc>(opApiFuncAddr);                                 \
-            auto api_ret = opApiFunc(workspace_addr, workspace_size, executor, acl_stream);                   \
-            TORCH_CHECK(api_ret == 0, "call " #aclnn_api " failed, detail:", aclGetRecentErrMsg());           \
-            ReleaseConvertTypes(converted_params);                                                            \
-            ReleaseHugeMem releaseMemFunc = reinterpret_cast<ReleaseHugeMem>(releaseMemAddr);                 \
-            if (releaseMemFunc) {                                                                             \
-                releaseMemFunc(nullptr, false);                                                               \
-            }                                                                                                 \
-            return api_ret;                                                                                   \
-        };                                                                                                    \
-        at_npu::native::OpCommand cmd;                                                                        \
-        cmd.Name(#aclnn_api);                                                                                 \
-        cmd.SetCustomHandler(acl_call);                                                                       \
-        cmd.Run();                                                                                            \
-        if (unInitMemFunc) {                                                                                  \
-            unInitMemFunc(nullptr, false);                                                                    \
-        }                                                                                                     \
+#define EXEC_NPU_CMD(aclnn_api, ...)                                                                              \
+    do {                                                                                                          \
+        static const auto getWorkspaceSizeFuncAddr = GetOpApiFuncAddr(#aclnn_api "GetWorkspaceSize");             \
+        static const auto opApiFuncAddr = GetOpApiFuncAddr(#aclnn_api);                                           \
+        static const auto initMemAddr = GetOpApiFuncAddr("InitHugeMemThreadLocal");                               \
+        static const auto unInitMemAddr = GetOpApiFuncAddr("UnInitHugeMemThreadLocal");                           \
+        static const auto releaseMemAddr = GetOpApiFuncAddr("ReleaseHugeMem");                                    \
+        TORCH_CHECK(getWorkspaceSizeFuncAddr != nullptr && opApiFuncAddr != nullptr, #aclnn_api, " or ",          \
+                    #aclnn_api "GetWorkspaceSize", " not in ", GetOpApiLibName(), ", or ", GetOpApiLibName(),     \
+                    "not found.");                                                                                \
+        auto acl_stream = c10_npu::getCurrentNPUStream().stream(false);                                           \
+        uint64_t workspace_size = 0;                                                                              \
+        uint64_t *workspace_size_addr = &workspace_size;                                                          \
+        aclOpExecutor *executor = nullptr;                                                                        \
+        aclOpExecutor **executor_addr = &executor;                                                                \
+        InitHugeMemThreadLocal initMemFunc = reinterpret_cast<InitHugeMemThreadLocal>(initMemAddr);               \
+        UnInitHugeMemThreadLocal unInitMemFunc = reinterpret_cast<UnInitHugeMemThreadLocal>(unInitMemAddr);       \
+        if (initMemFunc) {                                                                                        \
+            initMemFunc(nullptr, false);                                                                          \
+        }                                                                                                         \
+        auto converted_params = ConvertTypes(__VA_ARGS__, workspace_size_addr, executor_addr);                    \
+        static auto getWorkspaceSizeFunc = ConvertToOpApiFunc(converted_params, getWorkspaceSizeFuncAddr);        \
+        auto workspace_status = call(getWorkspaceSizeFunc, converted_params);                                     \
+        TORCH_CHECK(workspace_status == 0, "call " #aclnn_api " failed, detail:", aclGetRecentErrMsg());          \
+        void *workspace_addr = nullptr;                                                                           \
+        if (workspace_size != 0) {                                                                                \
+            at::TensorOptions options = at::TensorOptions(torch_npu::utils::get_npu_device_type());               \
+            auto workspace_tensor = at::empty({static_cast<int64_t>(workspace_size)}, options.dtype(c10::kByte)); \
+            workspace_addr = const_cast<void *>(workspace_tensor.storage().data());                               \
+        }                                                                                                         \
+        auto acl_call = [converted_params, workspace_addr, workspace_size, acl_stream, executor]() -> int {       \
+            typedef int (*OpApiFunc)(void *, uint64_t, aclOpExecutor *, const aclrtStream);                       \
+            OpApiFunc opApiFunc = reinterpret_cast<OpApiFunc>(opApiFuncAddr);                                     \
+            auto api_ret = opApiFunc(workspace_addr, workspace_size, executor, acl_stream);                       \
+            TORCH_CHECK(api_ret == 0, "call " #aclnn_api " failed, detail:", aclGetRecentErrMsg());               \
+            ReleaseConvertTypes(converted_params);                                                                \
+            ReleaseHugeMem releaseMemFunc = reinterpret_cast<ReleaseHugeMem>(releaseMemAddr);                     \
+            if (releaseMemFunc) {                                                                                 \
+                releaseMemFunc(nullptr, false);                                                                   \
+            }                                                                                                     \
+            return api_ret;                                                                                       \
+        };                                                                                                        \
+        at_npu::native::OpCommand cmd;                                                                            \
+        cmd.Name(#aclnn_api);                                                                                     \
+        cmd.SetCustomHandler(acl_call);                                                                           \
+        cmd.Run();                                                                                                \
+        if (unInitMemFunc) {                                                                                      \
+            unInitMemFunc(nullptr, false);                                                                        \
+        }                                                                                                         \
     } while (false)
 
 #endif  // PYTORCH_NPU_HELPER_HPP_

--- a/csrc/mla_preprocess/op_host/mla_preprocess.cpp
+++ b/csrc/mla_preprocess/op_host/mla_preprocess.cpp
@@ -673,8 +673,8 @@ std::tuple<at::Tensor &, at::Tensor &, at::Tensor &, at::Tensor &> mla_preproces
     // workspace
     uint64_t system_workspace_size = static_cast<uint64_t>(platformAscendC->GetLibApiWorkSpaceSize());
     uint64_t workspace_size = system_workspace_size + tilingData.userWorkspaceSize;
-    auto workspace_tensor =
-        at::empty({workspace_size}, at::TensorOptions().dtype(at::kByte).device(hiddenState.options().device()));
+    auto options = at::TensorOptions().dtype(at::kByte).device(hiddenState.options().device());
+    auto workspace_tensor = at::empty({static_cast<int64_t>(workspace_size)}, options);
 
     // tiling
     int32_t bIndex = N - 1;

